yklua/qsort.lua

```diff
@@ -0,0 +1,30 @@
+local function partition(arr, low, high)
+    local pivot = arr[high]
+    local i = low - 1
+
+    for j = low, high - 1 do
+        if arr[j] <= pivot then
+            i = i + 1
+            arr[i], arr[j] = arr[j], arr[i]
+        end
+    end
+
+    arr[i + 1], arr[high] = arr[high], arr[i + 1]
+    return i + 1
+end
+
+local function quicksort(arr, low, high)
+    if low < high then
+        local pi = partition(arr, low, high)
+        quicksort(arr, low, pi - 1)
+        quicksort(arr, pi + 1, high)
+    end
+end
+
+
+local arr = {7, 6, 5, 4, 3, 2, 1}
```

Comment:
That's a very short array! I suspect we will need something much longer to make the benchmark execute for a meaningful amount of time?

Reply:
ah yes, I just added it as an example.
How long is long enough?
I can change the array to something that runs for ~10 seconds with yklua using hwt

Reply:
That's with synchronised compilation, right? So 99% of that will be LLVM compiling things.

What I would do is run this benchmark under normal Lua until it runs for at least 0.1s and preferably nearer to 1s.

Benchmarks need to run for at least 0.1s

Reply:
yes, with a synchronised compilation

> What I would do is run this benchmark under normal Lua until it runs for at least 0.1s and preferably nearer to 1s.

ok, I'll do that

Reply:
In order to do that, I'd probably use a pseudo-random RNG with a fixed seed so that the array that's created always has the same contents. Admittedly that would become part of the benchmark too, but there's not much of a way around that.

Reply:
update 👉 60b070364a6b5c362eb4005f3d939407ce9355c3

Reply:
> but there's not much of a way around that.

We can hard-code the range value if we don't want its generation to affect the benchmark.

Reply:
Yes, we should hardcode a value that on your machine makes the benchmark take ~1s on normal (non-yk)lua.

Reply:
hardcoded 👉 a75125692f33acf4d186f09dfc6d41f90f91eca7